lib/src/handle_access_token_response.dart

```diff
@@ -2,114 +2,86 @@
 /// amount of time.
 const _expirationGrace = const Duration(seconds: 10);
 
-/// Returns a function that that parses parameters from a response with either a JSON or URL-encoded body.
-Function parseJsonOrUrlEncoded(void validate(bool condition, String message)) {
-  return (String contentTypeString, String body) {
-    var contentType = contentTypeString == null
-        ? null
-        : new MediaType.parse(contentTypeString);
-
-    // The spec requires a content-type of application/json, but some endpoints
-    // (e.g. Dropbox) serve it as text/javascript instead.
-    validate(contentType != null &&
-        (contentType.mimeType == "application/json" ||
-            contentType.mimeType == "text/javascript" ||
-            contentType.mimeType == "application/x-www-form-urlencoded"),
-        'content-type was "$contentType", expected "application/json" or "application/x-www-form-urlencoded"');
-
-    Map<String, dynamic> parameters;
-
-    if (contentType.mimeType == "application/x-www-form-urlencoded") {
-      parameters = {};
-
-      for (var unit in body.split('&')) {
-        var separator = unit.lastIndexOf('=');
-
-        // The '=' can't be the first or last character in a URL-encoded string
-        //
-        // For example, in 'a=b', the lowest index it can have is 1, and the greatest is
-        // `unit.length - 2`.
-        if (separator > 0 && separator < unit.length - 1) {
-          var key = unit.substring(0, separator);
-          var value = Uri.decodeComponent(unit.substring(separator + 1));
-          parameters[key] = value;
-        }
-      }
-    } else {
-      try {
-        var untypedParameters = JSON.decode(body);
-        validate(untypedParameters is Map,
-            'parameters must be a map, was "$parameters"');
-        parameters = DelegatingMap.typed(untypedParameters);
-      } on FormatException {
-        validate(false, 'invalid JSON');
-      }
-    }
-
-    return parameters;
-  };
-}
-
 /// Handles a response from the authorization server that contains an access
 /// token.
 ///
 /// This response format is common across several different components of the
 /// OAuth2 flow.
 /// 
 /// The scope strings will be separated by the provided [delimiter].
+///
+/// [getParameters] may be a function used to parse parameters out of responses from hosts that do not
+/// respond with application/json or application/x-www-form-urlencoded bodies. Use this
+/// to support authorization servers that do not completely follow the OAuth 2.0 standard.
 Credentials handleAccessTokenResponse(
     http.Response response,
     Uri tokenEndpoint,
     DateTime startTime,
     List<String> scopes,
     String delimiter,
-    {Map<String, dynamic> getParameters(String contentType, String body)}) {
-  if (response.statusCode != 200) _handleErrorResponse(response, tokenEndpoint);
+    {Map<String, dynamic> getParameters(String contentType, String body, Uri tokenEndpoint)}) {
```

Comment:
Why is this taking `tokenEndpoint` now?

Reply:
An error in my logic - it'll be removed within a few